### DISCUSSION TITLE
Parsing simple CSV

### DISCUSSION THREAD

**Discussion by kpym** [mmalcek/bafi#4] [Q&A] [answered]
I am trying to parse a very simple csv file named `test.csv`:
```csv
clatcoor
47.6669
49.0783
``` 

with the following command (under Windows):
```
bafi -f csv -i test.csv -t '?{{.clatcoor}}'
```
but I get the following error:
```
 writeStdout: template: new:1:2: executing "new" at <.clatcoor>: can't evaluate field clatcoor in type []map[string]inter
face {}
```
Any idea what is goning on ? This error message is note very user friendly, imo. It looks like go panic.

Another question : how can I tel bafi that my csv is using a different field separator ?

**Comment by mmalcek** [comment, marked as answer]
Hi in this case the issue is that you have range of values so you have to iterate over range - I've tested it and it works with 
```
.\bafi -f csv -i test.csv -t '?{{range .}}{{.clatcoor}}{{end}}'
```
I agree that better arror message would be nice, I'll enhance it in next versions

About separator - I will prepare some option to change separator in next version but currently it works only with CSV as specified in rfc4180 https://datatracker.ietf.org/doc/html/rfc4180

**Reply by kpym** [reply]
Thanks for you answer an your work !

**Reply by mmalcek** [reply]
Field separator can now be defined by additional parameter **-d** https://mmalcek.github.io/bafi/#command-line-arguments

**Reply by kpym** [reply]
Fantastic, thanks !
